src/VirtoCommerce.PricingModule.Data/Services/PricingServiceImpl.cs

```diff
@@ -211,10 +211,10 @@ public virtual async Task<Price[]> GetPricesByIdAsync(string[] ids)
             return result.Select(x => x.Clone() as Price).ToArray();
         }
 
-        public virtual async Task<PricelistAssignment[]> GetPricelistAssignmentsByIdAsync(string[] ids)
+        public virtual Task<PricelistAssignment[]> GetPricelistAssignmentsByIdAsync(string[] ids)
         {
             var cacheKey = CacheKey.With(GetType(), nameof(GetPricelistAssignmentsByIdAsync), string.Join("-", ids));
-            return await _platformMemoryCache.GetOrCreateExclusiveAsync(cacheKey, async cacheEntry =>
+            return _platformMemoryCache.GetOrCreateExclusiveAsync(cacheKey, async cacheEntry =>
```

Comment:
as there is async data loading, why these async/await we removed?

Reply:
AsyncFixer01 C# The method does not need to use async/await.
await is the last statement of the method so you can return Task and await in the caller method.